src/newtask/Form.js

```diff
@@ -0,0 +1,65 @@
+import { useRef, useState } from "react";
+
+const Form = (props) => {
+  const [toggleButton, setToggleButton] = useState(true);
+  const [data, setData] = useState({
+    taskName: "",
+    dueDate: "",
+    priority: "low",
+  });
+
+  const buttonEl = useRef();
+
+  const submitForm = (e) => {
+    e.preventDefault();
+    props.getNewTaskHandler(data);
+    props.closeFormHandler();
+  };
+
+  const getData = (e) => {
+    setData({ ...data, [e.target.name]: e.target.value });
```

Comment:
there seems to be a naming inconsistency here, the method is called `getData` but you're calling a `setData` function inside. I would suggest updating this.

Maybe as follows (please feel free to suggest otherwise):
```typescript
const updateTask = (event) => {
    const { target: { name, value } } = event
    setTask(previous => ({ ...previous, [name]: value }));
}
```

Please note that I took the liberty to update the method... restructuring the `name` and `value` from the event's `target` and renaming `e` to `event` (try to use descriptive variable names, always!)